### ISSUE TITLE
Incomplete fix of assertion violation at /src/smtsolvers/Debug.cc:85

### ISSUE THREAD

**Issue by rainoftime** [usi-verification-and-security/opensmt#79]
Related issue: https://github.com/usi-verification-and-security/opensmt/issues/60

----

Hi, for the following instance
~~~~
(set-logic QF_LIA)
(declare-fun _substvar_32_ () Bool)
(declare-fun v0 () Bool)
(declare-fun v2 () Bool)
(assert (or v2 v0))
(assert (distinct (or _substvar_32_ (or v2 v0)) (or v2 v0)))
(check-sat)
(push 1)
(check-sat)
(check-sat)
(check-sat)
(pop 1)
(check-sat)
~~~~

opensmt (commit 04c7fa3) throws an assertion violation
~~~~
unsat
unsat
unsat
unsat
unsatisfied clause: 4   3   ; (or v2 v0)
opensmt: /home/rainoftime/Work/tofuzz/opensmt/src/smtsolvers/Debug.cc:85: virtual void CoreSMTSolver::verifyModel(): Assertion `!failed' failed.
Aborted (core dumped)
~~~~

**Comment by rainoftime** [comment]
Another test case 
~~~~
(set-logic QF_LRA)
(declare-fun v0 () Bool)
(declare-fun v1 () Bool)
(push 1)
(assert false)
(check-sat)
(pop 1)
(assert (not (distinct (= (or (not v0) v1) v1) (not v1))))
(assert (or (not v0) v1))
(push 1)
(assert false)
(push 1)
(check-sat)
(pop 1)
(check-sat)
(pop 1)
(check-sat)
(push 1)
(check-sat)
(pop 1)
(check-sat)
~~~~

**Comment by blishko** [comment]
Fixed by #91
